model/flow/protocol_state.go

```diff
@@ -327,6 +327,16 @@ func (e *RichEpochProtocolStateEntry) CurrentEpochFinalView() uint64 {
 // EpochPhase returns the current epoch phase.
 // The receiver EpochProtocolStateEntry must be properly constructed.
 func (e *EpochProtocolStateEntry) EpochPhase() EpochPhase {
+	if e.EpochFallbackTriggered {
```

Comment:
I think it would be worth-while to add a comment here to call out the dependency on the Epoch Fallback state machine. just to reduce the risk of code changes in the future accidentally breaking the consistency between both components. 

```suggestion
	// CAUTION: the logic below that deduces the EpochPhase must be consistent with `epochs.FallbackStateMachine`,
	// which sets the fields we are using here. Specifically, we require that the FallbackStateMachine clears out 
	// any tentative values for a subsequent epoch _unless_ that epoch is already committed. 
	if e.EpochFallbackTriggered {
```